File: src/main/java/codes/qdbp/serverplugin/listeners/PlayerMoveListener.java
package codes.qdbp.serverplugin.listeners;

import codes.qdbp.serverplugin.Serverplugin;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;


public class PlayerMoveListener implements Listener {
    @EventHandler
    public void onPlayerMove(PlayerMoveEvent event) {
        Player player = event.getPlayer();
        if (Serverplugin.afkPlayerList.contains(player.getUniqueId())) {
            player.setInvulnerable(false);
            player.sendMessage("Du bist nicht mehr AFK");
            Serverplugin.afkPlayerList.remove(player.getUniqueId());
            Bukkit.getScheduler().cancelTask(Serverplugin.afkPlayerRunningTasksMap.get(player.getUniqueId()));
            Serverplugin.afkPlayerRunningTasksMap.remove(player.getUniqueId());
            player.playerListName(null);
        }
    }
}

File: src/main/kotlin/codes/qdbp/serverplugin/listeners/PlayerMoveListener.kt
package codes.qdbp.serverplugin.listeners

import codes.qdbp.serverplugin.misc.afkPlayers
import codes.qdbp.serverplugin.misc.endAfk
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerMoveEvent
import kotlin.time.DurationUnit
import kotlin.time.toDuration

class PlayerMoveListener : Listener {

    @EventHandler
    fun onPlayerMove(event: PlayerMoveEvent) {
        val player = event.player
        if (afkPlayers.containsKey(player.uniqueId)) {
            val time = afkPlayers[player.uniqueId]?.afkTime?.toDuration(DurationUnit.SECONDS)?.toComponents { hours, minutes, seconds, _ -> "${hours}h:${minutes}m:${seconds}s" } ?: ""
            player.sendMessage("Du warst $time afk.")
            player.endAfk()
        }
    }
}